app/src/main/java/com/p2p/presentation/truco/lobby/ConnectedPlayersTrucoAdapter.kt

```diff
@@ -0,0 +1,95 @@
+import com.p2p.R
+
+class ConnectedPlayersTrucoAdapter(
+    private val context: Context
+) :BaseAdapter() {
+
+    var players = listOf<String>()
+        set(value) {
+            field = value
+            notifyDataSetChanged()
+        }
+    var totalPlayers : Int = 4
+        set(value) {
+            field = value
+            notifyDataSetChanged()
+        }
+
+    override fun getItem(position: Int): Any {
+        return players[position]
+    }
+
+    override fun getItemId(position: Int): Long {
+        return position.toLong()
+    }
+
+    override fun getCount(): Int {
+        return if (players.count() <= totalPlayers) {
+            players.count()
+        } else {
+            totalPlayers
+        }
+    }
+
+    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
+        val layoutInflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
+        val connectedUserView: View = layoutInflater.inflate(R.layout.view_truco_connected_user, parent, false)
+        val player: String = players[position]
+        val playerName = connectedUserView.findViewById<View>(R.id.name) as TextView
+
+        playerName.text = player
+
+        if (position == 0) {
+            val imageView = connectedUserView.findViewById<ImageView>(R.id.avatar)
+            imageView.setImageResource(R.drawable.ic_baseline_account_circle_white)
+            connectedUserView.setBackgroundResource(R.drawable.grid_view_item_hand_player)
+        }
+
+        connectedUserView.setOnTouchListener { view, event ->
+            when (event.action) {
+                MotionEvent.ACTION_DOWN -> {
+                    val item = ClipData.Item(player as? CharSequence)
+                    val dragShadow = DragShadowBuilder(view!!)
+                    val dragData = ClipData(
+                        player as? CharSequence,
+                        arrayOf(ClipDescription.MIMETYPE_TEXT_PLAIN),
+                        item
+                    )
+
+                    if (position != 0) {
+                        view.background = ContextCompat.getDrawable(context, R.drawable.dotted_border)
+                    } else {
+                        view.background = ContextCompat.getDrawable(context, R.drawable.dotted_border)
+                        view.findViewById<ImageView>(R.id.avatar)
+                            .setImageResource(R.drawable.ic_baseline_account_circle_yellow)
+                    }
+
+                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+                        view.startDragAndDrop(dragData, dragShadow, null, 0)
+                    } else {
+                        view.startDrag(dragData, dragShadow, null, 0)
+                    }
+                }
+                MotionEvent.ACTION_UP -> view.performClick()
+            }
+            true
+        }
+
+        connectedUserView.setOnDragListener(DragListener())
+        return connectedUserView
```

Comment:
Podés usar view binding para facilitar las cosas y entiendo que tenés que usar el convertView si no es null:
```suggestion
        val userBinding = convertView
            ?.let { ViewTrucoConnectedUserBinding.bind(it) }
            ?: ViewTrucoConnectedUserBinding.inflate(LayoutInflater.from(context))
        val player: String = players[position]

        userBinding.name.text = player

        if (position == 0) {
            userBinding.avatar.setImageResource(R.drawable.ic_baseline_account_circle_white)
            userBinding.root.setBackgroundResource(R.drawable.grid_view_item_hand_player)
        }

        userBinding.root.setOnTouchListener { view, event ->
            when (event.action) {
                MotionEvent.ACTION_DOWN -> {
                    val item = ClipData.Item(player as? CharSequence)
                    val dragShadow = DragShadowBuilder(view!!)
                    val dragData = ClipData(
                        player as? CharSequence,
                        arrayOf(ClipDescription.MIMETYPE_TEXT_PLAIN),
                        item
                    )

                    if (position != 0) {
                        view.background = ContextCompat.getDrawable(context, R.drawable.dotted_border)
                    } else {
                        view.background = ContextCompat.getDrawable(context, R.drawable.dotted_border)
                        view.findViewById<ImageView>(R.id.avatar)
                            .setImageResource(R.drawable.ic_baseline_account_circle_yellow)
                    }

                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
                        view.startDragAndDrop(dragData, dragShadow, null, 0)
                    } else {
                        view.startDrag(dragData, dragShadow, null, 0)
                    }
                }
                MotionEvent.ACTION_UP -> view.performClick()
            }
            true
        }

        userBinding.root.setOnDragListener(DragListener())
        return userBinding.root
```

Reply:
OOOOH NICE, voy a probar a ver que onda, gracias!

Reply:
Che probé cambiandolo por el view binding, pero se me cagaba toda la vista (como que los cuadrados del gridview aparecían enormes), y como que tomaba mal los eventos de drag and drop y todo el cambio de colores.

No se que puede ser :c lo dejé como estaba antes